### ISSUE TITLE
warn

### ISSUE THREAD

**Issue by rkfsociety** [MiniGameWorlds/MiniGameWorld#21]
```yaml
[20:24:39 WARN]: org.bukkit.plugin.IllegalPluginAccessException: java.lang.ClassCastException: class org.bukkit.event.HandlerList cannot be cast to class org.bukkit.event.HandlerList (org.bukkit.event.HandlerList is in unnamed module of loader java.net.URLClassLoader @4c8151e8; org.bukkit.event.HandlerList is in unnamed module of loader java.net.URLClassLoader @484b61fc)
[20:24:39 WARN]: at org.bukkit.plugin.SimplePluginManager.getEventListeners(SimplePluginManager.java:709)
[20:24:39 WARN]: at org.bukkit.plugin.SimplePluginManager.registerEvent(SimplePluginManager.java:698)
[20:24:39 WARN]: at org.bukkit.plugin.SimplePluginManager.registerEvent(SimplePluginManager.java:668)
[20:24:39 WARN]: at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.listeners.CommonEventListener.registerAllEventListener(CommonEventListener.java:115)
[20:24:39 WARN]: at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.listeners.CommonEventListener.<init>(CommonEventListener.java:47)
[20:24:39 WARN]: at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.MiniGameWorldMain.registerEventListeners(MiniGameWorldMain.java:68)
[20:24:39 WARN]: at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.MiniGameWorldMain.onEnable(MiniGameWorldMain.java:45)
[20:24:39 WARN]: at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
[20:24:39 WARN]: at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[20:24:39 WARN]: at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
[20:24:39 WARN]: at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugin(CraftServer.java:563)
[20:24:39 WARN]: at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugins(CraftServer.java:477)
[20:24:39 WARN]: at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:727)
[20:24:39 WARN]: at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:503)
[20:24:39 WARN]: at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:313)
[20:24:39 WARN]: at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1202)
[20:24:39 WARN]: at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317)
[20:24:39 WARN]: at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by worldbiomusic** [comment]
@rkfsociety 
Thank you for notifying the plugin warnings.

But can you give me your bucket environment(spigot or paper, jdk version)?
I can't find warnings in spigot and paper bucket console.

---

You can also get help with setup plugins or any issues in [Discord](https://discord.com/invite/fJbxSy2EjA) or [Discussion](https://github.com/MiniGameWorlds/MiniGameWorld/discussions/categories/q-a).

**Comment by rkfsociety** [comment]
[Fri, 24. Dec 2021 17:58:06 MSK INFO] This server is running Paper version git-Paper-101 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: 464b171)
You are running the latest version

openjdk 17.0.1 2021-10-19
OpenJDK Runtime Environment (build 17.0.1+12-Ubuntu-121.04)
OpenJDK 64-Bit Server VM (build 17.0.1+12-Ubuntu-121.04, mixed mode, sharing)

**Comment by worldbiomusic** [comment]
It's probably due to a difference in JDK.

I'm using Oracle JDK and there are no warnings.

Can you change your JDK from open jdk to oracle jdk?

**Comment by rkfsociety** [comment]
java 17.0.1 2021-10-19 LTS
Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)

**Comment by rkfsociety** [comment]
[20:36:20] [Server thread/INFO]: [MiniGameWorld] Enabling MiniGameWorld v0.3.0
[20:36:20] [Server thread/INFO]: [MiniGameWorld] ================= MiniGameWorld =================
[20:36:30] [Server thread/WARN]: org.bukkit.plugin.IllegalPluginAccessException: java.lang.ClassCastException: class org.bukkit.event.HandlerList cannot be cast to class org.bukkit.event.HandlerList (org.bukkit.event.HandlerList is in unnamed module of loader java.net.URLClassLoader @7d0c8c9a; org.bukkit.event.HandlerList is in unnamed module of loader java.net.URLClassLoader @3b764bce)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.getEventListeners(SimplePluginManager.java:709)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.registerEvent(SimplePluginManager.java:698)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.registerEvent(SimplePluginManager.java:668)
[20:36:30] [Server thread/WARN]: 	at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.listeners.CommonEventListener.registerAllEventListener(CommonEventListener.java:115)
[20:36:30] [Server thread/WARN]: 	at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.listeners.CommonEventListener.<init>(CommonEventListener.java:47)
[20:36:30] [Server thread/WARN]: 	at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.MiniGameWorldMain.registerEventListeners(MiniGameWorldMain.java:68)
[20:36:30] [Server thread/WARN]: 	at MiniGameWorld-0.3.0.jar//com.worldbiomusic.minigameworld.MiniGameWorldMain.onEnable(MiniGameWorldMain.java:45)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugin(CraftServer.java:564)
[20:36:30] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugins(CraftServer.java:478)
[20:36:30] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:727)
[20:36:30] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:503)
[20:36:30] [Server thread/WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:313)
[20:36:30] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1202)
[20:36:30] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317)
[20:36:30] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by rkfsociety** [comment]
[20:41:05] [Server thread/INFO]: [MiniGameWorld] Disabling MiniGameWorld v0.3.0
[20:41:05] [Server thread/ERROR]: Error occurred while disabling MiniGameWorld v0.3.0 (Is it up to date?)
java.lang.NullPointerException: Cannot read the array length because the return value of "java.io.File.listFiles()" is null
	at com.worldbiomusic.minigameworld.managers.MiniGameManager.removeNotExistMiniGameData(MiniGameManager.java:326) ~[MiniGameWorld-0.3.0.jar:?]
	at com.worldbiomusic.minigameworld.MiniGameWorldMain.onDisable(MiniGameWorldMain.java:88) ~[MiniGameWorld-0.3.0.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:266) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:399) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:537) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:514) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_18_R1.CraftServer.disablePlugins(CraftServer.java:498) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.MinecraftServer.stopServer(MinecraftServer.java:1036) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.dedicated.DedicatedServer.stopServer(DedicatedServer.java:800) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.MinecraftServer.close(MinecraftServer.java:996) ~[paper-1.18.1.jar:git-Paper-101]
	at org.spigotmc.RestartCommand.shutdownServer(RestartCommand.java:92) ~[paper-1.18.1.jar:git-Paper-101]
	at org.spigotmc.RestartCommand.restart(RestartCommand.java:61) ~[paper-1.18.1.jar:git-Paper-101]
	at org.spigotmc.RestartCommand.restart(RestartCommand.java:40) ~[paper-1.18.1.jar:git-Paper-101]
	at org.spigotmc.RestartCommand$1.run(RestartCommand.java:31) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1560) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:480) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1470) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1264) ~[paper-1.18.1.jar:git-Paper-101]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.18.1.jar:git-Paper-101]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Comment by worldbiomusic** [comment]
Thanks for log.

It may be due to a folder structural change in 1.18 of paper.

Try with this [MiniGameWorld-0.3.0-rkfsociety-1.zip](https://github.com/MiniGameWorlds/MiniGameWorld/files/7775032/MiniGameWorld-0.3.0-rkfsociety-1.zip)

And contact me with discord to resolve the problem faster.

**Comment by worldbiomusic** [comment]
There will be update for paper1.18 folder structure change in next release.

**Comment by worldbiomusic** [comment]
Fixed in [Release 0.3.1](https://github.com/MiniGameWorlds/MiniGameWorld/releases/tag/0.3.1)
